diff --git a/top/maint.mk b/top/maint.mk
index e52deb4..f3e4ef9 100644
--- a/top/maint.mk
+++ b/top/maint.mk
@@ -1327,7 +1327,7 @@ announcement: NEWS ChangeLog $(rel-files)
 	    $$(case ,$(bootstrap-tools), in (*,gnulib,*)		\
 	       echo --gnulib-version=$(gnulib-version);; esac)		\
 	    --no-print-checksums					\
-	    $(addprefix --url-dir=, $(url_dir_list))
+	    $(addprefix --url-dir=, $(url_dir_list)) ~/announce-$(my_distdir)
 
 .PHONY: release-commit
 release-commit:
@@ -1428,7 +1428,6 @@ gl_noteworthy_news_ = * Noteworthy changes in release ?.? (????-??-??) [?]
 .PHONY: release-prep
 release-prep:
 	$(AM_V_GEN)$(MAKE) --no-print-directory -s announcement \
-	  > ~/announce-$(my_distdir)
 	$(AM_V_at)if test -d $(release_archive_dir); then	\
 	  ln $(rel-files) $(release_archive_dir);		\
 	  chmod a-w $(rel-files);				\
