--- gnulib/top/maint.mk	2012-07-07 18:15:13.000000000 +0700
+++ maint.mk	2012-07-14 17:20:08.000000000 +0700
@@ -1292,7 +1292,7 @@ announcement: NEWS ChangeLog $(rel-files)
 	    $$(case ,$(bootstrap-tools), in (*,gnulib,*)		\
 	       echo --gnulib-version=$(gnulib-version);; esac)		\
 	    --no-print-checksums					\
-	    $(addprefix --url-dir=, $(url_dir_list))
+	    $(addprefix --url-dir=, $(url_dir_list)) > ~/announce-$(my_distdir)
 
 .PHONY: release-commit
 release-commit:
@@ -1386,7 +1386,6 @@ release-prep:
 	$(AM_V_GEN)case $$RELEASE_TYPE in alpha|beta|stable) ;; \
 	  *) echo "invalid RELEASE_TYPE: $$RELEASE_TYPE" 1>&2; exit 1;; esac
 	$(AM_V_at)$(MAKE) --no-print-directory -s announcement \
-	  > ~/announce-$(my_distdir)
 	$(AM_V_at)if test -d $(release_archive_dir); then	\
 	  ln $(rel-files) $(release_archive_dir);		\
 	  chmod a-w $(rel-files);				\
