diff --git a/top/README-release b/top/README-release
index 29ba260..baaa7eb 100644
--- a/top/README-release
+++ b/top/README-release
@@ -1,35 +1,40 @@
-Here are most of the steps we (maintainers) follow when making a release.
+                      GNU Zile developers’ notes
+                      --------------------------
 
-* Start from a clean, up-to-date git directory on "master":
+     Copyright (c) 2011-2012 Free Software Foundation, Inc.
 
-    make -k maintainer-clean || { ./configure && make maintainer-clean; }
-    git checkout master
-    git pull origin master
+     Copying and distribution of this file, with or without
+     modification, are permitted in any medium without royalty
+     provided the copyright notice and this notice are preserved.
 
-* Ensure that the latest stable versions of autoconf, automake, etc.
-  are in your PATH.  See the buildreq list in bootstrap.conf for
-  the complete list of tools.
 
-* Ensure that you have no uncommitted diffs.  This should produce no
-  output:
+Automated releases
+------------------
 
-    git diff
+To make a release automatically, you need ncftp, and woger, from
+http://rrt.sc3d.org/Software/woger
 
-* Ensure that you've pushed all changes that belong in the release:
 
-    git push origin master
+GNU Style releases
+------------------
 
-* Check that the NixOS/Hydra autobuilder is reporting all is well:
+Here are most of the steps we (maintainers) follow when making a release.
 
-    http://hydra.nixos.org/jobset/gnu/@PACKAGE@-master
+* Ensure that the latest stable versions of autoconf, automake, etc.
+  are in your PATH.  See the buildreq list in bootstrap.conf for
+  the complete list of tools.
 
-* Run the following command to download any new translations:
+* Start from a clean, up-to-date git directory on "master", ensure
+  that you have no uncommitted diffs, have pushed all changes that
+  belong in the release, and that all tests pass:
 
-    ./bootstrap && ./configure
+    make "DISTCHECK_CONFIGURE_FLAGS=--enable-valgrind" prerelease
 
-* Pre-release testing: ensure that the following command succeeds:
+  (Make sure you have any needed Valgrind suppressions files.)
 
-    make check syntax-check distcheck
+* Check that the NixOS/Hydra autobuilder is reporting all is well:
+
+    http://hydra.nixos.org/jobset/gnu/@PACKAGE@-master
 
 * Set the date, version number, and release type [stable/alpha/beta] on
   line 3 of NEWS, commit that, and tag the release by running e.g.,
