diff --git a/top/README-release b/top/README-release
index 626777e..8e2d45a 100644
--- a/top/README-release
+++ b/top/README-release
@@ -1,3 +1,23 @@
+                      GNU Zile developers’ notes
+                      --------------------------
+
+     Copyright (c) 2011-2012 Free Software Foundation, Inc.
+
+     Copying and distribution of this file, with or without
+     modification, are permitted in any medium without royalty
+     provided the copyright notice and this notice are preserved.
+
+
+Automated releases
+------------------
+
+To make a release automatically, you need ncftp, and woger, from
+http://rrt.sc3d.org/Software/woger
+
+
+GNU Style releases
+------------------
+
 Here are most of the steps we (maintainers) follow when making a release.
 
 * Start from a clean, up-to-date git directory:
@@ -22,10 +42,11 @@ Here are most of the steps we (maintainers) follow when making a release.
 
     http://hydra.nixos.org/jobset/gnu/@PACKAGE@-master
 
-* Run "./bootstrap && ./configure".  This downloads any new translations.
+* Pre-release testing: ensure that the following command suceeds:
+
+    make "DISTCHECK_CONFIGURE_FLAGS=--enable-valgrind" check syntax-check distcheck
 
-* Pre-release testing: ensure that "make check syntax-check distcheck"
-  succeeds.
+  (Make sure you have any needed Valgrind suppressions files.)
 
 * Set the date, version number, and release type [stable/alpha/beta] on
   line 3 of NEWS, commit that, and tag the release by running e.g.,
